docs/architecture/cluster-registration.md

```diff
@@ -0,0 +1,25 @@
+# Cluster registry workflow
+
+The basic registration process is that the agent first sends a registration request to the hub cluster through the bootstrap token of the hub cluster. After the hub cluster detects the registration request, the hub cluster will start to create some necessary resources for registration. The registration request will be marked as approved after these resources to be created and ready.  When the cluster registration success, the agent will periodically send its heartbeat information to the hub cluster.
+
+## Architecture
+
+<img src="../images/cluster-registry-workflow.png" style="width:1000px;"/>
+
+## Workflow
+
+1. [Agent] Check if the `parent-cluster` exists.
+2. [Agent] Generate parent dedicated kubconfig
+   1. If the `parent-cluster` exists, use it.
+   2. If the `parent-cluster` does not exist, use bootstrap token.
+3. [Agent] Create `ClusterRegistrationRequest` with `step 2` kubeconfig, and wait for the request to be approved.
```

Comment:
* Create a `ClusterRegistrationRequest` in parent cluster using the `kubeconfig secret`, and wait for the request to be approved.